### PR TITLE
fileextractor: exclude extracted files by config file

### PR DESCRIPTION
**Pull request by disaykin**
I add the ability to exclude some files from extraction. Files is excluding by regular expressions on its path.

Regexps must be provided via option `--fileextractor-exlude-list <path to file with regular expressions>`.

Example file content:
```
^.+\.evtx$
^.+\.pf$
^.*\\windows\\system32\\spool\\drivers\\color\\.+\.icm$
```

### CODE REVIEW

**Comment by tklengyel** [comment]
Would be good to show an example on how to provide multiple regexps with this option. It may also make sense to provide this via a config file instead of the command line?

**Comment by disaykin** [comment]
@tklengyel I update description

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by disaykin** [comment]
I got "Argo Tunnel error" on Drakvuf CI results link

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
Seems to consistently break restarting drakvuf on a win7 sp1 x64 VM.

File: src/plugins/fileextractor/fileextractor.cpp [hunk 1]
```diff
@@ -216,6 +220,9 @@ event_response_t fileextractor::createfile_ret_cb(drakvuf_t,
             addr_t file = 0;
             auto filename = get_file_name(vmi, info, handle, &file, nullptr);
 
+            if (is_in_exclude_list(filename))
+                return VMI_EVENT_RESPONSE_NONE;
```
Comment (BonusPlay) [inline src/plugins/fileextractor/fileextractor.cpp:224]: shouldn't you call `createfile_ret_hooks.erase(hook_id);` here?

**Review by BonusPlay — COMMENTED** [review]
regarding CI fails it's probably drakvuf-CI

**Comment by tklengyel** [comment]
> regarding CI fails it's probably drakvuf-CI

Not so sure. There are intermittent failures on the CI for sure but when a PR repeatedly fails when the main branch runs fine, it's likely the PR.

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

File: src/plugins/fileextractor/fileextractor.cpp (hunk 1 above)
Comment (disaykin) [reply src/plugins/fileextractor/fileextractor.cpp:224]: You are right. Thanks!

Comment (disaykin) [reply src/plugins/fileextractor/fileextractor.cpp]: Fixed

**Comment by disaykin** [comment]
@tklengyel The reason for the failure of the tests was the incorrect path to the file with the list of system calls in the test.

I remove this change (and it fixed the test):
``` 
            case 'S':
                if (!std::filesystem::exists(optarg))
                {
                    fprintf(stderr, "file %s does not exist!\n", optarg);
                    return drakvuf_exit_code_t::FAIL;
                }
```

**Pull request merged**